src/gc/gc.cpp

```diff
@@ -838,7 +831,7 @@ class t_join
             return TRUE;
         }
 
-        if (Interlocked::Decrement(&join_struct.r_join_lock) != (join_struct.n_threads - 1))
+        if (Interlocked::CompareExchange(&join_struct.r_join_lock, 0, join_struct.n_threads) == 0)
```

Comment:
@sdmaclea, just to clarify, did you mean a single CompareExchange in general can be significantly faster than a single Exchange or Decrement on arm64, or that in total it can be faster just because we are only doing the compare but not the store in many cases whereas Exchange/Decrement always does the store?